src/lib.rs

```diff
@@ -83,54 +92,104 @@ mod window_counter;
 pub mod connectors;
 pub mod resolvers;
 
+/// Uniquely identifies a claim
+#[derive(Copy, Clone, Debug)]
+pub(crate) struct ClaimId(pub u64);
+
+impl ClaimId {
+    fn new() -> Self {
+        let id = usdt::UniqueId::new().as_u64();
```

Comment:
I'm fine with these being "basically random" - we're only really using them for instrumentation - but the changes in this PR "make the claim value once" and use it, rather than making it dynamically around probes from start -> done like we did before.